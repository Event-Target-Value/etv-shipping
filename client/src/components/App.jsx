import React from 'react';
import Modal from './ModalOne.jsx';
import Zipcode from './Zipcode.jsx';
import axios from 'axios';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      product: {},
      currentZip: 94112,
      show: false,
      zipMatch: false,
      showZip: false
    };
    this.closeModal = this.closeModal.bind(this);
    this.getProduct = this.getProduct.bind(this);
    this.setProduct = this.setProduct.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.showZipcode = this.showZipcode.bind(this);
    this.closeZipcode = this.closeZipcode.bind(this);
    this.setDefaultZip = this.setDefaultZip.bind(this);
  }

  componentDidMount() {
    // this.getProduct(Math.floor(Math.random() * (101 - 1) + 1));
    this.getProduct(45);
  }

  handleError(error) {
    console.log(error)
  }

  getProduct(id) {
    //origin / shipping/findproduct
    axios.get(`/shipping/findProduct/${id}`)
    .then(this.setProduct)
    .catch(this.handleError)
  }

  setProduct({data}) {
    this.setState({product: data});
    if (data.zip.includes(this.state.currentZip)) {
      this.setState({zipMatch: true})
    }
  };

  closeModal = (e) => {
    this.setState({
      show: false
    });
  };

  showModal = (e) => {
    this.setState({
      show: true
    });
  };

  showZipcode = (e) => {
    e.preventDefault();
    this.setState({
      showZip: true
    });
  };

  closeZipcode = (e) => {
    e.preventDefault();
    this.setState({
      showZip:false
    });
  };

  setDefaultZip = (e) => {
    e.preventDefault();
    this.setState({
      currentZip: 94112,
      showZip: false
    });
    if ((this.state.product.zip).includes(94112)) {
      this.setState({zipMatch: true})
    } else {
      this.setState({zipMatch: false})
    }
  };

  handleSubmit(e) {
    e.preventDefault();
    if (isNaN(parseInt((document.getElementById("zipcode").value).slice(0,5))) ||
      ((document.getElementById("zipcode").value)).length < 5) {
      alert("Please enter a 5 digit zip code")
    } else {
    let newZip = parseInt((document.getElementById("zipcode").value).slice(0,5));
    this.setState({currentZip: newZip, showZip: false});
      if ((this.state.product.zip).includes(newZip)) {
        this.setState({zipMatch: true})
      } else {
        this.setState({zipMatch: false})
      }
    };
    // this.setState({currentZip: parseInt((event.target.value))})
  };

  render () {
    const zmatch = this.state.zipMatch;
    return (
      <div className="App">
        <div className = "NonModal">
          <div className = "zip">
          <span style = {{color: 'green'}}>Deliver to </span>
          {this.state.currentZip} </div>
          <button className = "showZip" id = "showZip" onClick = {e => {
            this.showZipcode(e)
          }}><u>Edit zip code</u></button>
          <Zipcode setDefaultZip = {this.setDefaultZip}
          handleSubmit = {this.handleSubmit} showZip = {this.state.showZip} closeZipcode = {this.closeZipcode}></Zipcode>
          {zmatch ? <button className = "showModal" id="showModal" onClick = {e => {
            this.showModal(e)
          }}> Ship it </button> :
          <div>
          <div className = "notAvailable">not available</div>
          <div className = "sorry">Sorry, this can't be shipped to your zip code.</div>
          </div>}
        </div>
        {this.state.show && (
        <Modal id="modal" className = "ModalOne" product={this.state.product} show={this.state.show} closeModal={this.closeModal}>
        </Modal>
        )}
      </div>)
  }
}

export default App;