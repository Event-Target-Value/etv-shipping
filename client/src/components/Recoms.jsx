import React from "react";

const Recoms= (props) => (
  <div className = "singleProduct">
  <img className="recomImage" src={props.oneRecom.recImage}></img>
  <span>${props.oneRecom.price}</span>
  <span>{props.oneRecom.name}</span>
  </div>
)


// Overflow-x: hidden 
//  Element.scroll() 
// React refs
// z-index


//refactor Recoms to > Recommendations

export default Recoms;