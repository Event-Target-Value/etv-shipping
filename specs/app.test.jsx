import React from 'react';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import App from '../client/src/components/App.jsx';
import Modal from '../client/src/components/ModalOne.jsx';
import Recoms from '../client/src/components/Recoms.jsx';

describe('<App />', () => {
    it('React App functional and correctly rendering', () => {
      const wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Renders the default zipcode of 94112', () => {
      const wrapper = shallow(<App />);
      expect(wrapper.find('.zip').text()).toEqual("Deliver to 94112")
    })

    it('Renders the zipcode based on submission', () => {
      const wrapper = shallow(<App />);
      expect(wrapper.find('.zip').text()).toEqual("Deliver to 94112")
    })
});



// describe('<Modal />', () => {
//   it('Modal functional and correctly rendering', () => {
//    const wrapper = shallow(<Modal />);
//    expect(wrapper).toMatchSnapshot();
//   });
// });





