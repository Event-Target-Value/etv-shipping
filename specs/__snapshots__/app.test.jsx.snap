// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> React App functional and correctly rendering 1`] = `
<div
  className="App"
>
  <div
    className="NonModal"
  >
    <h1>
      PRODUCT SHIPPING COMPONENT
    </h1>
    <div
      className="zip"
    >
      Deliver to 
      94112
    </div>
    <button
      className="showZip"
      id="showZip"
      onClick={[Function]}
    >
      Edit zip code
    </button>
    <Zipcode
      closeZipcode={[Function]}
      handleSubmit={[Function]}
      showZip={false}
    />
    <button
      className="showModal"
      id="showModal"
      onClick={[Function]}
    >
       Ship it 
    </button>
  </div>
  <Modal
    className="ModalOne"
    closeModal={[Function]}
    product={Object {}}
    show={false}
  />
</div>
`;
